src/test/SmartComponents/CreateImageWizard/CreateImageWizard.test.js

```diff
@@ -1,26 +1,173 @@
+    const back = screen.getByRole('button', { name: /Back/ });
+    const cancel = screen.getByRole('button', { name: /Cancel/ });
+
+    return [ next, back, cancel ];
+}
+
+function verifyCancelButton(cancel) {
+    cancel.click();
+
+    // this goes back to the landing page
+    waitForElement(
+        () => [
+            screen.getByRole('heading', { name: /Image Builder/ }),
+            screen.getByText('Create a new image'),
+        ],
+        { screen }
+    );
+}
+
+describe('Create Image Wizard', () => {
     beforeEach(() => {
         renderWithReduxRouter(<CreateImageWizard />);
     });
+
     test('renders component', () => {
         // check heading
         screen.getByRole('heading', { name: /Create a new image/ });
 
         // left sidebar navigation
-        const nav = screen.getByRole('navigation');
-        getByText(nav, 'Release');
-        getByText(nav, 'Target environment');
-        getByText(nav, 'Registration');
-        getByText(nav, 'Review');
-
-        // buttons
-        screen.getByRole('button', { name: /Next/ });
+        const sidebar = screen.getByRole('navigation');
+
+        getByText(sidebar, 'Release');
+        getByText(sidebar, 'Target environment');
+        getByText(sidebar, 'Registration');
+        getByText(sidebar, 'Review');
+    });
+});
+
+describe('Step Release', () => {
+    beforeEach(() => {
+        renderWithReduxRouter(<CreateImageWizard />);
+
+        // left sidebar navigation
+        const sidebar = screen.getByRole('navigation');
+        const anchor = getByText(sidebar, 'Release');
+
+        // load from sidebar
+        anchor.click();
+    });
+
+    test('clicking Next loads Target environment', () => {
+        const [ next, , ] = verifyButtons();
+        next.click();
+
+        screen.getByText('Destination');
+        screen.getByText('Secret access key');
+    });
+
+    // todo: Should we have the Back button here ?
+    // Maybe not b/c it doesn't lead anywhere
+
+    test('clicking Cancel loads landing page', () => {
+        const [ , , cancel ] = verifyButtons();
+        verifyCancelButton(cancel);
+    });
+});
+
+describe('Step Target environment', () => {
+    beforeEach(() => {
+        renderWithReduxRouter(<CreateImageWizard />);
+
+        // left sidebar navigation
+        const sidebar = screen.getByRole('navigation');
+        const anchor = getByText(sidebar, 'Target environment');
+
+        // load from sidebar
+        anchor.click();
+    });
+
+    test('clicking Next loads Registration', () => {
+        const [ next, , ] = verifyButtons();
+        next.click();
+
+        screen.getByText('Register the system');
+    });
+
+    test('clicking Back loads Release', () => {
+        const [ , back, ] = verifyButtons();
+        back.click();
+
+        // note: there doesn't seem to be anything better on this screen
+        screen.getByText('Red Hat Enterprise Linux (RHEL) 8.2');
```

Comment:
Any idea if I can do better here ?

Reply:
Hm is it possible to somehow get it via the `value` attribute? Alternatively I'd also be ok if it were just a check to make sure the `Select` is there.

Reply:
I've added data-testid attribute and used that b/c the react testing library doesn't allow selecting elements by their ID. FTR they have philosophical opinion as to why that is which just makes our job a bit more miserable.